core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -192,6 +201,32 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
       </a>
     </span> ++
     <div id="application-timeline" class="collapsed">
+      {
+        if (MAX_TIMELINE_JOBS < jobs.size) {
+          <div>
+            <strong>
+              This page has more than the maximum number of jobs that can be shown in the
+              visualization! Only the most recent {MAX_TIMELINE_JOBS} submitted/completed jobs
+              (of {jobs.size} total) are shown.
```

Comment:
Nit: how about just
```
Only the most recent {MAX_TIMELINE_JOBS} submitted jobs(of {jobs.size} total) are shown. 
```
Not sure about whether we should hint the user to tune the configuration.

Reply:
Thanks. This sentence is brought from `StagePage.scala` so I've modified it too.